packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -513,6 +513,487 @@ export class CloudFormationDeleteStackAction extends CloudFormationDeployAction
+   *         "ParameterKey": "KeyName",
+   *         "ParameterValue": "true"
+   *     }
+   * ]
+   *
+   * This cannot be used in conjunction with parameters.
+   *
+   * @default No parameters will be used (unless parameters is used)
+   */
+  readonly parametersPath?: codepipeline.ArtifactPath;
+
+  /**
+   * Indicates that the template can create and update resources, depending on the types of resources in the template.
+   *
+   * You must use this property if you have IAM resources in your stack template or you create a stack directly from a template containing macros.
+   *
+   * @default The StackSet will have no IAM capabilities.
+   */
+  readonly cfnCapabilities?: Array<cdk.CfnCapabilities>;
+
+  /**
+   * Determines how IAM roles are created and managed. If the field is not specified, the default is used. For information, see Permissions models for
+   * stack set operations.
+   *
+   * Note
+   * This parameter can only be changed when no stack instances exist in the stack set.
+   *
+   * @default SELF_MANAGED
+   */
+  readonly permissionModel?: StackSetPermissionModel;
+
+  /**
+   * Note
+   * Because AWS CloudFormation StackSets performs operations across multiple accounts, you must define the necessary permissions in those accounts
+   * before you can create the stack set.
+   *
+   * Note
+   * This parameter is optional for the SELF_MANAGED permissions model and is not used for the SERVICE_MANAGED permissions model.
+   *
+   * The IAM role in the administrator account used to perform stack set operations.
+   *
+   * If you do not specify the role, it is set to AWSCloudFormationStackSetAdministrationRole. If
+   * you specify ServiceManaged, you must not define a role name.
+   *
+   * @default AWSCloudFormationStackSetAdministrationRole
+   */
+  readonly administrationRole?: iam.IRole;
+
+  /**
+   * Note
+   * Because AWS CloudFormation StackSets performs operations across multiple accounts, you must define the necessary permissions in those accounts
+   * before you can create the stack set.
+   *
+   * Note
+   * This parameter is optional for the SELF_MANAGED permissions model and is not used for the SERVICE_MANAGED permissions model.
+   *
+   * The name of the IAM role in the target accounts used to perform stack set operations. The name may contain alphanumeric characters, any of the
+   * following characters: _+=,.@-, and no spaces. The name is not case sensitive. This role name must be a minimum length of 1 character and maximum
+   * length of 64 characters. Role names must be unique within the account. The role name specified here must be an existing role name. Do not specify
+   * this role if you are using customized execution roles. If you do not specify the role name, it is set to AWSCloudFormationStackSetExecutionRole.
+   * If you set Service_Managed to true, you must not define a role name.
+   *
+   * @default AWSCloudFormationStackSetExecutionRole
+   */
+  readonly executionRoleName?: string;
```

Comment:
Same comment here as in `parameters`. We have 3 properties (`administrationRole`, `executionRoleName` and `permissionModel`) that have a relationship. We should substitute them with one property that encompasses that relationship in a way that makes it safe at compile-time.